nvflare/tool/job/config/configer.py

```diff
@@ -146,7 +161,26 @@ def merge_configs(
                             if key not in key_indices:
                                 # not every client has app_config, app_script
                                 if key not in [APP_SCRIPT_KEY, APP_CONFIG_KEY]:
-                                    raise ValueError(f"Invalid config key: '{key}' for file '{file}'")
+                                    if key.startswith(".") or key.endswith("."):
+                                        raise ValueError(f"invalid key {key} for file {file}")
+
+                                parent, index, key = split_array_key(key)
+                                if parent is not None and index is not None:
+                                    parent_config_list = config.get(parent)
+                                    if not isinstance(parent_config_list, list):
+                                        raise ValueError(f"invalid key '{key}' for file {file}")
+                                    index_config = parent_config_list[index]
+                                    index_config.put(key, cli_value)
+                                    key_value = index_config.get(key)
+                                else:
+                                    config.put(key, cli_value)
+                                    key_value = config.get(key)
+
+                                tokens = key.split(".")
+                                if len(tokens) > 1:
+                                    key = tokens[-1]
+                                key_index = KeyIndex(key, key_value)
+                                key_indices[key] = [key_index]
```

Comment:
For situations like:
`-f xxx.conf args.model.length=100 args.length=50`
The first key is "args.model.length" and the key "length" will be put inside the key_indices dict.

The second key is "args.length" and the key "length" will be put inside the key_indices dict, replacing the value of the first one?

Reply:
thanks for the example, then we can enhance the display later.